corehq/apps/hqcase/api.py

```diff
@@ -182,11 +181,8 @@ def _get_bulk_updates(domain, all_data, user):
 
 
 def _missing_cases(domain, case_ids):
-    return set(case_ids) - {
-        case.case_id for case in
-        CaseAccessors(domain).get_cases(case_ids)
-        if case.domain == domain
-    }
+    real_case_ids = CaseAccessorSQL.get_case_ids_that_exist(domain, case_ids)
```

Comment:
This will not work for couch domains. Would it make sense to add a new `get_case_ids_that_exist` db accessor that is implemented for both Couch and SQL? I think that might be what @snopoke originally had in mind.

Reply:
Thanks for bringing this up - I should've mentioned it in the PR description.  The (upcoming) list view of the API uses the CaseSearch back-end, which is only enabled for a subset of domains.  I was planning to do the initial release of this behind a feature flag, and only support SQL and CaseSearch domains.  There are efforts to migrate all domains to both of those.

Supporting couch domains wouldn't be hard though, so I could be persuaded.  If not I suppose a view decorator encoding these assumptions would be a good idea.  What do you think?

Reply:
I wish I could say "We will have all domains migrated off of Couch by the time this is released generally" but given how long it's taken I'm hesitant. It does seem easy to add support for Couch (possibly not as efficient as SQL, but I think that doesn't matter), so that seems like the best route. I expect to have all of the very large domains migrated off of Couch very soon, in case that's worth anything.

Reply:
alright, added in support for couch.